crates/services/p2p/src/p2p_service.rs

```diff
@@ -517,13 +562,23 @@ impl FuelP2PService {
         event: FuelBehaviourEvent,
     ) -> Option<FuelP2PEvent> {
         match event {
-            FuelBehaviourEvent::Gossipsub(event) => self.handle_gossipsub_event(event),
+            FuelBehaviourEvent::Gossipsub(event) => {
+                self.update_libp2p_metrics(&event);
+                self.handle_gossipsub_event(event)
+            }
             FuelBehaviourEvent::PeerReport(event) => self.handle_peer_report_event(event),
```

Comment:
Do we want to track peer report, request response, heartbeat?

Just curios. We can leave it for follow-up.

Reply:
we probably should, but libp2p does not provide any metrics from their end for these :)